features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/store-new/source/src/app/components/Apis/Listing/StarRatingBar.jsx

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Link } from 'react-router-dom';
+import { withStyles } from '@material-ui/core/styles';
+import Cancel from '@material-ui/icons/Cancel';
+import { StarRate } from '@material-ui/icons';
+import Alert from '../../Shared/Alert';
+import Api from '../../../data/api';
+import AuthManager from '../../../data/AuthManager';
+import StarRatingSummary from '../Details/StarRatingSummary';
+
+
+/**
+ *
+ *
+ * @param {*} theme
+ */
+const styles = theme => ({
+    starRate: {
+        fontSize: 30,
+        color: theme.custom.starColor,
+    },
+    noStarRate: {
+        fontSize: 30,
+        color: theme.palette.grey.A200,
+    },
+    removeRating: {
+        fontSize: 20,
+        color: 'black',
+    },
+    userRating: {
+        display: 'flex',
+        justifyContent: 'flex-start',
+        alignItems: 'center',
+    },
+});
+
+/**
+ *
+ *
+ * @class StarRatingBar
+ * @extends {React.Component}
+ */
+class StarRatingBar extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            rating: {
```

Comment:
If we don't have conflicting key names , shall we move the `rating` object items into `state` level.

i:e

```javascript
this.state = {
                avgRating: 0,
                userRating: 0,
                count: 0
}

So then we could easily spread the state when updating, No nested objects
```

Reply:
Fixed